### PR TITLE
Rich message

### PR DESCRIPTION
**Pull request by reytum**

### CODE REVIEW

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/AlCustomizationSettings.java [hunk 1]
```diff
@@ -587,6 +588,14 @@ public void setFilterGallery(Map<String, Boolean> filterGallery) {
         this.filterGallery = filterGallery;
     }
 
+    public boolean isEnableShareConversation() {
+        return enableShareConversation;
```
Comment (mukeshthawani) [inline kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/AlCustomizationSettings.java]: @reytum What can we share by enabling it? An attachment? 
A description would be great!

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java [hunk 2]
```diff
@@ -3978,44 +3993,76 @@ public void hideAwayMessage(Message message) {
...
-                sendMessage((String) object);
+                sendMessage((String) object, getStringMap(replyMetadata));
                 break;
 
             case AlRichMessage.SEND_HOTEL_DETAILS:
-                sendHotelDetailMessage((AlHotelBookingModel) object);
+                sendHotelDetailMessage((AlHotelBookingModel) object, getStringMap(replyMetadata));
                 break;
 
             case AlRichMessage.SEND_ROOM_DETAILS_MESSAGE:
```
Comment (mukeshthawani) [inline kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java]: At some point, we should remove these `Hotel` related cases and use the generic template Ids.

File: kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/AlCustomizationSettings.java (hunk 1 above)
Comment (reytum) [reply kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/AlCustomizationSettings.java]: This is for internal use only, for Agent app, to share the conversation url

File: kommunicateui/src/main/res/menu/mobicom_basic_menu_for_normal_message.xml [hunk 3]
```diff
@@ -132,4 +132,12 @@
         android:showAsAction="collapseActionView"
         android:title="@string/logout"
         android:visible="false" />
+
+    <item
+        android:id="@+id/share_conversation"
+        android:orderInCategory="12"
+        android:showAsAction="never"
+        android:title="Share"
```
Comment (Sunilkumarr) [inline kommunicateui/src/main/res/menu/mobicom_basic_menu_for_normal_message.xml]: Move `Share`  string file

**Pull request merged**